operator/src/main/java/io/streamzi/eventflow/ClientCache.java

```diff
@@ -34,9 +37,27 @@
         apiClients.put("local", osClient);
 
         //Try and find a Strimzi installation in this namespace
-        List<Service> strimziServices = osClient.services().inNamespace(osClient.getNamespace()).withLabel("strimzi.io/name").list().getItems();
-        if (strimziServices.size() > 0) {
-            bootstrapServerCache.put("local", strimziServices.get(0).getMetadata().getName() + "." + osClient.getNamespace() + ".svc.cluster.local:9092");
+        final CustomResourceDefinition kafkaCRD = osClient.customResourceDefinitions().withName("kafkas.kafka.strimzi.io").get();
+        if (kafkaCRD == null) {
+            logger.info("Can't find Kafka CRD - Not setting BOOTSTRAP_SERVERS");
+        } else {
+            List<Kafka> kafkas = osClient.customResources(
+                    kafkaCRD,
+                    Kafka.class,
+                    KafkaAssemblyList.class,
+                    DoneableKafka.class)
+                    .inNamespace(osClient.getNamespace())
+                    .list()
+                    .getItems();
+
+            if (kafkas.size() > 0) {
+                String localBs = kafkas.get(0).getMetadata().getName() + "-kafka-bootstrap." + kafkas.get(0).getMetadata().getNamespace() + ".svc.cluster.local:9092";
+                logger.info("Setting BOOTSTRAP_SERVERS for local to " + localBs);
+                bootstrapServerCache.put("local", localBs);
```

Comment:
cache is currently only a simple map, right ?

Reply:
Yes, is that ok?

Reply:
> Yes, is that ok?

yes, we just need to have that in mind - I have a history w/ _improving_ caches :)